From b9e10f09dce88c579ae9d847b27568cd1d6814fa Mon Sep 17 00:00:00 2001
From: Gary Zhang <b13634@freescale.com>
Date: Wed, 11 Jul 2012 10:15:42 +0800
Subject: ENGR00209905 WM8962: support for continuously playback diff sample
 bit streams

support for continuously playback different sample bit audio
streams with -Dplughw:0,0 option
such as the command: 'aplay -Dplughw:0,0 16bit.wav 24bit.wav'
before prohibit reenter hw_params, now remove this limitation
to support this feature.

Signed-off-by: Gary Zhang <b13634@freescale.com>
---
 sound/soc/codecs/wm8962.c  |    6 ------
 sound/soc/imx/imx-wm8962.c |    7 ++-----
 2 files changed, 2 insertions(+), 11 deletions(-)

diff --git a/sound/soc/codecs/wm8962.c b/sound/soc/codecs/wm8962.c
index 2f5ce28..190720e 100644
--- a/sound/soc/codecs/wm8962.c
+++ b/sound/soc/codecs/wm8962.c
@@ -3159,9 +3159,6 @@ static int wm8962_hw_params(struct snd_pcm_substream *substream,
 	int aif0 = 0;
 	int adctl3 = 0;
 
-	if (codec->dapm.bias_level != SND_SOC_BIAS_OFF)
-		return 0;
-
 	wm8962->bclk = snd_soc_params_to_bclk(params);
 	if (params_channels(params) == 1)
 		wm8962->bclk *= 2;
@@ -3463,9 +3460,6 @@ static int wm8962_set_fll(struct snd_soc_codec *codec, int fll_id, int source,
 	if (fll_div.theta || fll_div.lambda)
 		fll1 |= WM8962_FLL_FRAC;
 
-	/* Stop the FLL while we reconfigure */
-	snd_soc_update_bits(codec, WM8962_FLL_CONTROL_1, WM8962_FLL_ENA, 0);
-
 	snd_soc_update_bits(codec, WM8962_FLL_CONTROL_2,
 			    WM8962_FLL_OUTDIV_MASK |
 			    WM8962_FLL_REFCLK_DIV_MASK,
diff --git a/sound/soc/imx/imx-wm8962.c b/sound/soc/imx/imx-wm8962.c
index aa3122f..76299a6 100644
--- a/sound/soc/imx/imx-wm8962.c
+++ b/sound/soc/imx/imx-wm8962.c
@@ -98,9 +98,6 @@ static int imx_hifi_hw_params(struct snd_pcm_substream *substream,
 	u32 dai_format;
 	unsigned int pll_out;
 
-	if (codec->dapm.bias_level != SND_SOC_BIAS_OFF)
-		return 0;
-
 	dai_format = SND_SOC_DAIFMT_I2S | SND_SOC_DAIFMT_NB_NF |
 		SND_SOC_DAIFMT_CBM_CFM;
 
@@ -128,8 +125,8 @@ static int imx_hifi_hw_params(struct snd_pcm_substream *substream,
 	else
 		pll_out = sample_rate * 256;
 
-	ret = snd_soc_dai_set_pll(codec_dai, WM8962_FLL_MCLK,
-				  WM8962_FLL_MCLK, priv->sysclk,
+	ret = snd_soc_dai_set_pll(codec_dai, WM8962_FLL_OSC,
+				  WM8962_FLL_OSC, priv->sysclk,
 				  pll_out);
 	if (ret < 0)
 		pr_err("Failed to start FLL: %d\n", ret);
-- 
1.7.9.5

